src/raft.c

```diff
@@ -599,6 +637,40 @@ static void clearClientSessions(RedisRaftCtx *rr)
     rr->client_session_dict = RedisModule_CreateDict(rr->ctx);
 }
 
+static void timeoutBlockedCommand(RedisRaftCtx *rr, raft_entry_t *entry)
+{
+    RedisModule_Assert(entry->data_len == sizeof(raft_index_t));
+    raft_index_t idx;
+    memcpy(&idx, entry->data, sizeof(idx));
+
+    BlockedCommand *bc = getBlockedCommand(idx);
+    if (!bc) {
+        /* unblock handler called before timeout was applied */
+        return;
+    }
+
+    if (RedisModule_CallReplyPromiseAbort(bc->reply, NULL) != REDISMODULE_OK) {
+        /* shouldn't happen with normal redis commands */
+        LOG_WARNING("timeoutBlockedCommand: failed to abort %s at %lu", bc->command, idx);
+        return;
+    }
+
+    if (bc->req) {
+        switch (getNullReplyType(bc->command)) {
+            case REDISRAFT_NULL_ARRAY:
+                RedisModule_ReplyWithNullArray(bc->req->ctx);
+                break;
+            default:
+                RedisModule_ReplyWithNull(bc->req->ctx);
+                break;
+        }
```

Comment:
```suggestion
        if ((strncasecmp(bc->command, "bzpopmin", 8) == 0) ||
            (strncasecmp(bc->command, "bzpopmax", 8) == 0) ||
            (strncasecmp(bc->command, "bzmpop", 6) == 0)) {
            RedisModule_ReplyWithNullArray(bc->req->ctx);
        } else {
            RedisModule_ReplyWithNull(bc->req->ctx);
        }

```

I think we can move that function here to keep logic local. Also, this way we can avoid defining things this way.

Reply:
so, the reason I didn't go to this right now is that I'm thinking of a general metadata parsing, so that we don't have to strncasecmp all the time for them.  With that said, can do that later.